### PR TITLE
Added the support to change the log level.

### PR DESCRIPTION
**Pull request by nitin-deamon**
Added the support change log level dynamically. Xdatamove  will use this api to change log level.

### CODE REVIEW

File: common/logger.go [hunk 1]
```diff
@@ -159,6 +160,14 @@ func (jl *jobLogger) ShouldLog(level pipeline.LogLevel) bool {
 	return level <= jl.minimumLevelToLog
 }
 
+func (jl *jobLogger) ChangeLogLevel(level pipeline.LogLevel) bool {
```
Comment (linuxsmiths) [inline common/logger.go:163]: ChangeLogLevel() need not return bool. It can be a void function.

File: jobsAdmin/JobsAdmin.go [hunk 2]
```diff
@@ -470,6 +470,16 @@ func (ja *jobsAdmin) LogToJobLog(msg string, level pipeline.LogLevel) {
 	ja.JobLogger.Log(pipeline.LogWarning, prefix+msg) // use LogError here, so that it forces these to get logged, even if user is running at warning level instead of Info.  They won't have "warning" prefix, if Info level was passed in to MessagesForJobLog
 }
 
+func (ja *jobsAdmin) ChangeLogLevel(level pipeline.LogLevel, jobId common.JobID) (bool, error) {
```
Comment (linuxsmiths) [inline jobsAdmin/JobsAdmin.go:473]: again, this can just return error

File: ste/mgr-JobMgr.go [hunk 3]
```diff
@@ -97,14 +97,16 @@ type IJobMgr interface {
...
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
 func NewJobMgr(concurrency ConcurrencySettings, jobID common.JobID, appCtx context.Context, cpuMon common.CPUMonitor, level common.LogLevel,
-	       commandString string, logFileFolder string, tuner ConcurrencyTuner,
-	       pacer PacerAdmin, slicePool common.ByteSlicePooler, cacheLimiter common.CacheLimiter, fileCountLimiter common.CacheLimiter,
-	       jobLogger common.ILoggerResetable) IJobMgr {
+	commandString string, logFileFolder string, tuner ConcurrencyTuner,
+	pacer PacerAdmin, slicePool common.ByteSlicePooler, cacheLimiter common.CacheLimiter, fileCountLimiter common.CacheLimiter,
+	jobLogger common.ILoggerResetable) IJobMgr {
```
Comment (linuxsmiths) [inline ste/mgr-JobMgr.go:109]: Undo unwanted changes. 
There are few more blank space changes.

File: ste/mgr-JobMgr.go [hunk 4]
```diff
@@ -182,13 +184,17 @@ func NewJobMgr(concurrency ConcurrencySettings, jobID common.JobID, appCtx conte
...
+
 	go jm.reportJobPartDoneHandler()
 	go jm.handleStatusUpdateMessage()
-	
+
 	return &jm
 }
 
+func (jm *jobMgr) ChangeLogLevel(level pipeline.LogLevel) bool {
+	return jm.logger.ChangeLogLevel(level)
```
Comment (linuxsmiths) [inline ste/mgr-JobMgr.go:195]: better to guard it with

if jm.logger {
}

since it may technically be nil

File: common/logger.go [hunk 5]
```diff
@@ -159,6 +160,14 @@ func (jl *jobLogger) ShouldLog(level pipeline.LogLevel) bool {
...
+	if level == pipeline.LogNone {
+		return false
+	}
+	jl.minimumLevelToLog = level
```
Comment (linuxsmiths) [inline common/logger.go:167]: Now that we are writing to this variable, we can have multiple goroutines call this and technically have issues.
Though we may be calling it only once, but better to call out.

// This update is not necessarily safe from multiple goroutines simultaneously calling it.
// Typically we will call ChangeLogLevel() once at the beginning so it should be ok.

File: ste/mgr-JobMgr.go (hunk 3 above)
Comment (nitin-deamon) [reply ste/mgr-JobMgr.go:109]: Its gofmt, change whenever i save on IDE it moved it.

**Pull request merged**